forms-flow-web/src/apiManager/services/userservices.js

```diff
@@ -23,3 +26,57 @@ export const updateUserlang = (data) => {
       });
   };
 };
+
+export const getFormioRoleIds = (...rest) => {
+  // eslint-disable-next-line
+  const done = rest.length ? rest[0] : () => {};
+  let data;
+  if (MULTITENANCY_ENABLED) {
+    data = localStorage.getItem("tenantData");
+  } else {
+    data = localStorage.getItem("roleIds");
+  }
+  // if data is there no need to call api
+  if (data) {
+    return (dispatch) => {
+      if (MULTITENANCY_ENABLED) {
+        dispatch(setTenantData(JSON.parse(data)));
+        data = JSON.parse(data.form);
+      } else {
+        data = JSON.parse(data);
+      }
+      dispatch(setRoleIds(data));
+      dispatch(setAccessForForm(data));
+      done(null, data);
+    };
+  } else {
+    let url = MULTITENANCY_ENABLED ? API.GET_TENANT_DATA : API.FORMIO_ROLES;
+
+    return (dispatch) => {
+      httpGETRequest(url, {}, UserService.getToken(), true)
+        .then((res) => {
+          if (res.data) {
+            localStorage.setItem("roleIds", JSON.stringify(res.data.form));
+            dispatch(setRoleIds(res.data?.form));
```

Comment:
Both Multitenancy and normal casethe roleids appear in data.form?

Reply:
Check error case /add condition or ?. All cases for checking the result is obtained correctly

Reply:
yes